### PR TITLE
chore: verify with 4.6

### PR DESCRIPTION
**Pull request by ajewellamz**
_Issue #, if available:_

_Description of changes:_

_Squash/merge commit message, if applicable:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: project.properties [hunk 1]
```diff
@@ -5,4 +5,5 @@
 #     export `export `cat ./aws-cryptographic-material-providers-library-dafny/project.properties`
 # The Java project include this file as a Gradle properties
 # And the Dotnet projects include and parse this file.
-dafnyVersion=4.2.0
\ No newline at end of file
+dafnyVersion=4.2.0
+dafnyVerifyVersion=4.2.0
```
Comment (texastony) [inline project.properties:9]: Question: Do we intend to bump this to `4.6.0`?

Comment (ajewellamz) [reply project.properties:9]: We wanted to, but StandardLibrary/UTF8.ValidUTF8Concat doesn't verify in 4.6.0, and it's not clear what to do about it. It currently verifies under nightly, and so once 4.7 is available we'll try again.

File: .github/workflows/dafny_verify_version.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# This workflow reads the project.properties
+# into the environment variables
+# and then creates an output variable for `dafnyVersion`
```
Comment (seebees) [inline .github/workflows/dafny_verify_version.yaml:3]:
```suggestion
# and then creates an output variable for `dafnyVerifyVersion `
```

**Review by seebees — COMMENTED** [review]
I think just update the name in the workflow

**Review by seebees — APPROVED** [review]
LGTM

**Pull request merged**